pkg/controller/vitesscell/reconcile_topo.go

```diff
@@ -61,6 +61,14 @@ func (r *ReconcileVitessCell) reconcileTopology(ctx context.Context, vtc *planet
 	}
 	defer ts.Close()
 
+	if *vtc.Spec.TopologyReconciliation.PruneSrvKeyspaces {
+		return r.pruneSrvKeyspaces(ctx, vtc, keyspaces, ts)
+	}
+
+	return nil, nil
```

Comment:
Even if we don't prune, we still need to build and return the `topoEntries` list. Let's fetch `srvKeyspaceNames` here, compute `topoEntries` and `wanted`, and then pass `srvKeyspaceNames` and `wanted` into the prune subroutine.

Reply:
Refactored per our offline discussion. When you have a moment PTAL.